Content.Server/GameObjects/Components/Power/AME/AMEPartComponent.cs

```diff
@@ -37,6 +38,11 @@ async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs args)
                 var mapGrid = _mapManager.GetGrid(args.ClickLocation.GetGridId(_serverEntityManager));
                 var tile = mapGrid.GetTileRef(args.ClickLocation);
                 var snapPos = mapGrid.SnapGridCellFor(args.ClickLocation, SnapGridOffset.Center);
+                if (mapGrid.GetSnapGridCell(snapPos, SnapGridOffset.Center).Where(sc => sc.Owner.HasComponent<AMEShieldComponent>()).Count() > 0)
+                {
+                    Owner.PopupMessage(args.User, Loc.GetString("Shielding is already there!"));
+                    return true;
+                }
 
                 var ent = _serverEntityManager.SpawnEntity("AMEShielding", mapGrid.GridTileToLocal(snapPos));
                 ent.Transform.LocalRotation = Owner.Transform.LocalRotation;
```

Comment:
I have some comments on this but @Zumorica I'm pretty sure AMEPart should be replaced with a construction graph instead.

Reply:
this makes sense, but also won't solve the whole overplacing issue
though that's something the construction system needs to solve anyway

Reply:
That is true.

I'm fine to merge as is, just slap a TODO on it or make an issue for the construction part and make an issue for the overlapping.